volatility3/framework/plugins/linux/sockstat.py

```diff
@@ -0,0 +1,506 @@
+
+        Returns:
+            unix_sock: The kernel's `unix_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        unix_sock = sock.cast("unix_sock")
+        state = unix_sock.get_state()
+        saddr = unix_sock.get_name()
+        sinode = unix_sock.get_inode()
+        if unix_sock.peer != 0:
+            peer = unix_sock.peer.dereference().cast("unix_sock")
+            daddr = peer.get_name()
+            dinode = peer.get_inode()
+        else:
+            daddr = dinode = ""
+
+        saddr_tag = f"{saddr} {sinode}"
+        daddr_tag = f"{daddr} {dinode}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return unix_sock, sock_stat
+
+    def _inet_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_INET/6 socket families
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            inet_sock: The kernel's `inet_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        inet_sock = sock.cast("inet_sock")
+        saddr = inet_sock.get_src_addr()
+        sport = inet_sock.get_src_port()
+        daddr = inet_sock.get_dst_addr()
+        dport = inet_sock.get_dst_port()
+        state = inet_sock.get_state()
+
+        if inet_sock.get_family() == "AF_INET6":
+            saddr = f"[{saddr}]"
+
+        saddr_tag = f"{saddr}:{sport}"
+        daddr_tag = f"{daddr}:{dport}"
+        sock_stat = saddr_tag, daddr_tag, state
+        return inet_sock, sock_stat
+
+    def _netlink_sock(self, sock: objects.StructType) -> Tuple[objects.StructType, Tuple[str, str, str]]:
+        """Handles the AF_NETLINK socket family
+
+        Args:
+            sock: Kernel generic `sock` object
+
+        Returns:
+            netlink_sock: The kernel's `netlink_sock` object
+            sock_stat: A tuple with the source, destination and state strings.
+        """
+        netlink_sock = sock.cast("netlink_sock")
+
+        saddr_list = []
+        src_portid = f"portid:{netlink_sock.portid}"
+        saddr_list.append(src_portid)
+        if netlink_sock.groups != 0:
+            groups_bitmap = netlink_sock.groups.dereference()
+            groups_str = f"groups:0x{groups_bitmap:08x}"
+            saddr_list.append(groups_str)
+
+        daddr_list = []
+        dst_portid = f"portid:{netlink_sock.dst_portid}"
+        daddr_list.append(dst_portid)
+        dst_group = f"group:0x{netlink_sock.dst_group:08x}"
+        daddr_list.append(dst_group)
+        module = netlink_sock.module
+        if module and netlink_sock.module.name:
+            module_name_str = utility.array_to_string(netlink_sock.module.name)
+            module_name = f"lkm:{module_name_str}"
+            daddr_list.append(module_name)
+
+        saddr_tag = ",".join(saddr_list)
+        daddr_tag = ",".join(daddr_list)
```

Comment:
Really not so keen on this, maybe in the run/generate method, but not inside a helper function please, return a list, flatten it when the data is definitely not going to be used by anything else.